scripts/helm/publish-chart-yaml.sh

```diff
@@ -10,15 +10,89 @@
 # are changed to the appTag.
 # If the appTag is a stable release then both version and appVersions are changed to the appTag.
 
+# Write output to error output stream.
+echo_stderr() {
+  echo -e "${1}" >&2
+}
+
 die()
 {
   local _return="${2:-1}"
-  echo "$1" >&2
+  echo_stderr "$1"
   exit "${_return}"
 }
 
 set -euo pipefail
 
+# Requires a 'git fetch origin' (origin being the remote entry for openebs/mayastor-extensions) or equivalent.
+latest_release_branch() {
+  cd "$ROOTDIR"
+
+  local remote_url=""
+  local openebs_remote=""
+  for remote in $(git remote show); do
+    remote_url=$(git ls-remote --get-url $remote)
+    if [[ "$remote_url" =~ ([\/:]openebs\/mayastor-extensions)$ ]]; then
+      openebs_remote=$remote
+      break
+    fi
+  done
+  if [ -z "$openebs_remote" ]; then
+    die "failed to find git remote pointing to 'openebs/mayastor-extensions'"
+  fi
+  # The latest release branch name is required for generating the helm chart version/appVersion
+  # for the 'main' branch only.
+  # The 'git branch' command in the below lines checks remote refs for release/x.y branch entries.
+  # Because the 'main' branch is not a significant branch for a user/contributor, this approach towards
+  # finding the latest release branch assumes that this script is used when the 'openebs/mayastor-extensions'
+  # repo is present amongst git remote refs. This happens automatically when the 'openebs/mayastor-extensions'
+  # repo is cloned, and not a user/contributor's fork.
+  local latest_release_branch=$(git branch \
+    --all \
+    --list "*${openebs_remote}/release/*.*" \
+    --format '%(refname:short)' \
+    --sort 'refname' \
+    | tail -n 1)
+
+  cd - >/dev/null
+
+  echo "${latest_release_branch#*$openebs_remote/}"
+}
+
+main_branch_chart_app_version_from_latest_release_branch_name() {
```

Comment:
This is quite long...
```suggestion
main_branch_version_from_release_branch() {
```

Reply:
Renamed this function